homeassistant/components/synology_dsm/config_flow.py

```diff
@@ -407,32 +407,31 @@ async def async_step_backup_share(
         if not self.saved_user_input:
             self.saved_user_input = user_input
 
-        if CONF_BACKUP_PATH not in user_input and CONF_BACKUP_SHARE not in user_input:
-            return self.async_show_form(
-                step_id="backup_share",
-                data_schema=vol.Schema(
-                    {
-                        vol.Required(CONF_BACKUP_SHARE): SelectSelector(
-                            SelectSelectorConfig(
-                                options=[
-                                    SelectOptionDict(value=s.path, label=s.name)
-                                    for s in self.shares
-                                ],
-                                mode=SelectSelectorMode.DROPDOWN,
-                            ),
-                        ),
-                        vol.Required(
-                            CONF_BACKUP_PATH,
-                            default=f"{DEFAULT_BACKUP_PATH}_{slugify(self.hass.config.location_name)}",
-                        ): str,
-                    }
-                ),
-            )
-
-        user_input = {**self.saved_user_input, **user_input}
-        self.saved_user_input = {}
+        if CONF_BACKUP_PATH in user_input:
```

Comment:
```suggestion
        if user_input:
```

Reply:
Then we should change that. We shouldn't pass invalid user input to a step method. The user input of a step method is the user input from the form of the step method or should at least correspond to that format. Data coming from other steps should be stored on the instance and accessed that way.

Reply:
Yeah, nothing in this PR is required for the February release.